### ISSUE TITLE
False-positive: if() with continue/break should be considered connected to looping

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#145]
I have the following code:

```php
foreach ([] as $object) {
    if (mt_rand() >= 0.1) {
        break;
    }
}
```

I receive the warning "statement could be decoupled from foreach". you can note that the `if()` is _not directly_ connected to looping, but that it could controls the looping by `break` (works too with `continue`).

**Comment by kalessil** [comment]
The statement really does not operate on the loop variables.
I'd suggest suppressing or adjusting if: `if (null !== $object && mt_rand() >= 0.1)`.

**Comment by kalessil** [comment]
@rentalhost I have to decline the ticket, to keep the inspection as simple as possible.

**Comment by rentalhost** [comment]
@kalessil I guess that you can check if an internal structure (`if()`) have some `break`/`continue`, does not? I belive that you will need do something similar on #144. It could be done just after the problem is detected, but before register it (then only will use more resource if it pass by "current phases"). I think that your solution is a bit workaround, once that my `$object` could be a non-null (in this case is preferible I just set a `@noinspection`).

**Comment by kalessil** [comment]
Hm, `break/continue/throw/return` check wouldn't hurt much I guess - It wort giving it a try at least. Thanks for pointing to it, I was looking onto condition itself only and missed you point

And yes, my proposals are workarounds - just wait for a fix and new dev-build in the repo!

**Comment by kalessil** [comment]
@rentalhost : I pushed a fresh binary, so you can give a try to the fixed version.

**Comment by rentalhost** [comment]
@kalessil Wow! I really thought the code was much bigger than that. Awesome solution! Works very well for me! Thanks a lot!
